### ISSUE TITLE
Is it possible to have 2 outputs from the same node linked to a node that has 2 inputs?

### ISSUE THREAD

**Issue by rrcarpenter** [Fattorino/ImNodeFlow#29]
I think this part of the code is why the second out pin is not calling _m_val = m_behaviour()_

```cpp
 template<class T>
    const T &OutPin<T>::val()
    {
        if (std::find((*m_inf)->get_recursion_blacklist().begin(), (*m_inf)->get_recursion_blacklist().end(), m_parent->getUID()) == (*m_inf)->get_recursion_blacklist().end())
        {
            (*m_inf)->get_recursion_blacklist().emplace_back(m_parent->getUID());
            m_val = m_behaviour();
        }

        return m_val;
    }
```

![grafik](https://github.com/user-attachments/assets/8cf6184f-0e25-4a7e-bf00-193e7a3feedd)

**Comment by Fattorino** [comment]
Fixed with commit [10553cf](https://github.com/Fattorino/ImNodeFlow/commit/10553cf5ab9ac421f40a49eedb3787552e45b76d)

**Comment by rrcarpenter** [comment]
On line 490 in the header file this should be changed to 

```cpp
std::vector<PinUID> m_nodeRecursionBlacklist;
```

Also if there are nodes with the same out pin name this will not work.
Maybe do something like using the parent uid and the out uid:

```cpp
template<class T>
    const T &OutPin<T>::val()
    {
          PinUID po_uid = m_parent->getUID() + m_uid;   // po -> parent + out pin uid
          if (std::find((*m_inf)->get_recursion_blacklist().begin(), (*m_inf)->get_recursion_blacklist().end(), po_uid) == (*m_inf)->get_recursion_blacklist().end())
        {
            (*m_inf)->get_recursion_blacklist().emplace_back(po_uid);
            m_val = m_behaviour();
        }
        return m_val;
    }
```
